src/nft/utils/bag.ts

```diff
@@ -0,0 +1,81 @@
+import { BagItem, BagItemStatus, BagStatus, RoutingItem, UpdatedGenieAsset } from 'nft/types'
+
+import { compareAssetsWithTransactionRoute } from './txRoute/combineItemsWithTxRoute'
+import { filterUpdatedAssetsByState } from './updatedAssets'
+
+export function getPurchasableAssets(itemsInBag: BagItem[]): UpdatedGenieAsset[] {
+  return itemsInBag.filter((item) => item.status !== BagItemStatus.UNAVAILABLE).map((item) => item.asset)
+}
+
+function createBagFromUpdatedAssets(
+  unavailable: UpdatedGenieAsset[],
+  priceChanged: UpdatedGenieAsset[],
+  unchanged: UpdatedGenieAsset[]
+): BagItem[] {
+  return [
+    ...unavailable.map((unavailableAsset) => ({
+      asset: unavailableAsset,
+      status: BagItemStatus.UNAVAILABLE,
+    })),
+    ...priceChanged.map((changedAsset) => ({
+      asset: changedAsset,
+      status: BagItemStatus.REVIEWING_PRICE_CHANGE,
+    })),
+    ...unchanged.map((unchangedAsset) => ({
+      asset: unchangedAsset,
+      status: BagItemStatus.REVIEWED,
+    })),
+  ]
+}
+
+function evaluateNextBagState(
+  hasAssets: boolean,
+  shouldReview: boolean,
+  hasAssetsInReview: boolean,
+  shouldRefetchCalldata: boolean
+): { nextBagStatus: BagStatus; lockBag: boolean } {
+  if (!hasAssets) {
+    return { nextBagStatus: BagStatus.ADDING_TO_BAG, lockBag: false }
+  }
+
+  if (shouldReview) {
+    if (hasAssetsInReview) {
+      return { nextBagStatus: BagStatus.IN_REVIEW, lockBag: false }
+    }
+
+    return { nextBagStatus: BagStatus.CONFIRM_REVIEW, lockBag: false }
+  }
+
+  if (shouldRefetchCalldata) {
+    return { nextBagStatus: BagStatus.CONFIRM_QUOTE, lockBag: false }
+  }
+
+  return { nextBagStatus: BagStatus.CONFIRMING_IN_WALLET, lockBag: true }
```

Comment:
Are there cases where the BagStatuses and lockBag state differ from what's here? You could create an enum/map mapping status to lockBag state so you don't have to track them separately.

Reply:
crazier how none of this is actually necessary when one state only returns false lmao